src/decode/decode.spec.ts

```diff
@@ -491,9 +271,29 @@ describe('decode', () => {
     'should decode item %#',
     { timeout: 10_000 }, // some cases take more than 5 seconds to run
     ({ bytes, expected }) => {
```

Comment:
Now all test cases are run for 1 of 4 scenarios:
- as-is
- self-described (this covers the deleted test cases above)
- with offset view (this covers the new tests to cover the fix)
- self-described with offset view (this covers the new tests to cover the fix)